### PR TITLE
Fix segfault in parallel CN gradient

### PR DESCRIPTION
**Pull request by marvinfriede**

### CODE REVIEW

File: src/dftd4/ncoord.f90 [hunk 1]
```diff
@@ -168,8 +168,8 @@ subroutine ncoord_derf(mol, trans, cutoff, rcov, en, cn, dcndr, dcndL)
    dcndL(:, :, :) = 0.0_wp
    cutoff2 = cutoff**2
 
-   !$omp parallel do default(none) reduction(+:cn, dcndr, dcndL) &
-   !$omp shared(mol, trans, cutoff2, rcov, en) &
+   !$omp parallel do schedule(runtime) default(none) &
```
Comment (TyBalduf) [inline src/dftd4/ncoord.f90:171]: I'm guessing that making this `schedule(runtime)` leads to a default of dynamic scheduling when `OMP_SCHEDULE` isn't specified (this is at least the case for gcc, see [section 3.12](https://gcc.gnu.org/onlinedocs/gcc-9.3.0/libgomp.pdf)), while the default for no schedule specified is static scheduling.

I'm still not entirely sure why that fixes the segfault, but it presumably has OpenMP related storage on the heap for dynamic scheduling and on the stack for static scheduling.

So maybe this should just be `schedule(dynamic)` if the real fix is just `schedule(runtime)` defaulting to dynamic scheduling. Or alternatively, just removing the reduction entirely when looping over the number of atoms, as awvwgk suggested.

Comment (marvinfriede) [reply src/dftd4/ncoord.f90:171]: How would one remove the reduction? By creating local arrays for each OMP thread?

@awvwgk You mentioned that there is already an example in dftd3, where the reduction is removed. I wasn't able to find it. Where is it?

Comment (awvwgk) [reply src/dftd4/ncoord.f90:171]: In dftd3 we don't reduce the dcndr / dcndL, but avoid computation of them at all.

https://github.com/dftd3/simple-dftd3/blob/bb8fd834df3de0134d3045ad7103fa67e39b6577/src/dftd3/disp.f90#L102-L103

Alternatives are to use atomic updates

https://github.com/tblite/tblite/blob/ba9c91e8708b332fad67bcb021c7054b544ef7fb/src/tblite/coulomb/charge/effective.f90#L365-L368

Or to allocate temporary arrays for each thread

https://github.com/tblite/tblite/blob/ba9c91e8708b332fad67bcb021c7054b544ef7fb/src/tblite/coulomb/charge/effective.f90#L560-L562

Comment (marvinfriede) [reply src/dftd4/ncoord.f90:171]: I copied and adapted the s-dftd3 version. The example in the issue that first raised this problem now runs without problems.

**Review by awvwgk — APPROVED** [review]
Looks good to me now

**Pull request merged**